service/service.go

```diff
@@ -407,10 +407,20 @@ func (s *service) UploadImageToStorage(ctx context.Context, imageBase64Str strin
 	return imageId, nil
 }
 
-func (s *service) UpsertProductCategory(ctx context.Context, prodCategory *prodEntity.ProductCategory) error {
-	existingProdCategory, err := s.productRepository.GetProductCategoryByName(ctx, strings.ToLower(prodCategory.Name))
-	if err != nil {
-		return err
+func (s *service) UpsertProductCategory(ctx context.Context, prodCategory *prodEntity.ProductCategory) (err error) {
+	var existingProdCategory *prodEntity.ProductCategory
+	switch prodCategory.ID {
```

Comment:
how many categories do we have for the product mas?
and if we use **switch case**, it is better if we have a specific value for each category.
 I think we can use if instead:
example:
```
if prodCategory.ID == 0 {
   // get product by name
} else {
   // get product by id
}
```